### PR TITLE
Clean up launch of Vision Glass

### PR DESCRIPTION
**Pull request by felipeerias**
Structure the code so it responds to changes in connected devices, USB permissions, etc.

Implement OnConnectionListener to get notified when the device is connected and disconnected.

Receive notifications when the USB permission has been granted.

Try to ensure that the state of the app remains consistent.

Add debug logs.

### CODE REVIEW

File: app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java [hunk 1]
```diff
@@ -113,54 +116,94 @@ private void runPhoneUICallback(PhoneUIButtonsCallback callback) {
...
+            return;
+        }
+
+        if (!VisionGlass.getInstance().isConnected()) {
+            Log.d(LOGTAG, "Glasses not connected yet");
+            return;
+        }
+
+        if (VisionGlass.getInstance().hasUsbPermission()) {
+            mIsAskingForPermission = false;
```
Comment (svillar) [inline app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java:177]: I think we should initialize it to false and remove this from here. Perhaps you'd like to reset it to false in the USB permission request handler.

File: app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java [hunk 2]
```diff
@@ -250,12 +298,16 @@ public boolean onGenericMotionEvent(MotionEvent aEvent) {
...
         Log.d(LOGTAG, "PlatformActivity onPause");
         super.onPause();
-        synchronized (mRenderLock) {
-            queueRunnable(activityPausedRunnable);
-            try {
-                mRenderLock.wait();
-            } catch(InterruptedException e) {
-                Log.e(LOGTAG, "activityPausedRunnable interrupted: " + e.toString());
+
+        // TODO This check is needed to prevent a crash when pausing before 3D mode has started.
```
Comment (svillar) [inline app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java:302]: Perhaps the "TODO" is not needed?

**Review by svillar — CHANGES_REQUESTED** [review]
Looking great. Perhaps too much debugging for my taste but I think it's OK for this initial phase.

I just have a comment, see inline

File: app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java (hunk 1 above)
Comment (felipeerias) [reply app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java]: Fixed.

File: app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java (hunk 2 above)
Comment (felipeerias) [reply app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java]: Fixed.

**Pull request merged**